### DISCUSSION TITLE
`flip` does not perform repositioning for floating elements

### DISCUSSION THREAD

**Discussion by piscopancer** [floating-ui/floating-ui#3109] [React Native]
Hi, what am I doing wrong?

As I scroll down the scrollview I want my tooltip to prefer `top` when the element appears, change its place to `bottom` when I scroll so far there's no place above the parent element. It does not happen so. I must be missing smth. See the video please and the code below

white element is a some sort of tooltip, "look at me" is a referenced element

thanks in advance

```tsx
  const fl = useFloating({
    placement: 'top', // prefer top, I guess
    middleware: [flip()], 
  })

  return (
    <ScrollView overScrollMode='never' {...fl.scrollProps}>
      {/*  unrelated elements */}
      <View className='bg-orange-500/10 p-4 h-[200vh] justify-center'>
        <View>
          <View ref={fl.refs.setReference} className='p-4 bg-orange-500/50 text-center'>
            <Text>Look at me</Text>
          </View>
          <View ref={fl.refs.setFloating} style={fl.floatingStyles} className='bg-white rounded-full w-full py-12'>
            <Text className='text-center'>💣</Text>
          </View>
        </View>
      </View>
``` 

https://github.com/user-attachments/assets/cd0c766c-d50d-4c7e-91d6-7fb709e86814

**Comment by atomiks** [comment]
Is it repositioning (rerendering) on scroll? With `scrollProps`, you may need `sameScrollView: false`

**Reply by atomiks** [reply]
I notice you're running Android. The new release may fix it, specifically the `getClippingRect` calculation: https://github.com/floating-ui/floating-ui/pull/3107
